### PR TITLE
Expression to identify if the pipeline is remote or inline

### PR DESCRIPTION
**Pull request by sarthakkasat**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 1]
```diff
@@ -739,6 +739,18 @@ The list of stages selected for execution.
 
 The pipeline level delegate selectors selected via runtime input.  
 
+### <+pipeline.storeType>
+
+If the pipeline is stored in harness the expression will resolve to Inline, if the pipeline is stored in Git repository the expression will resolve to remote.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-variables.md:744]:
```suggestion
If the pipeline is stored in Harness, the expression resolves to inline. If the pipeline is stored in a Git repository, the expression resolves to remote.
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 2]
```diff
@@ -739,6 +739,18 @@ The list of stages selected for execution.
...
+
+### <+pipeline.repoName>
+
+For Remote pipeline the expression will resolve to Git repository name. For Inline pipeline the expression will resolve to null.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-variables.md:748]:
```suggestion
For remote pipelines, the expression resolves to the Git repository name. For inline pipelines, the expression resolves to null.
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 3]
```diff
@@ -739,6 +739,18 @@ The list of stages selected for execution.
...
+
+### <+pipeline.branchName>
+
+For Remote pipeline the expression will resolve to Git branch in which pipeline is present. For Inline pipeline the expression will resolve to null.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-variables.md:752]:
```suggestion
For remote pipelines, the expression resolves to the Git branch where the pipeline exists. For inline pipelines, the expression will resolve to null.
```

**Review by brian-f-harness — APPROVED** [review]

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 4]
```diff
@@ -739,6 +739,18 @@ The list of stages selected for execution.
 
 The pipeline level delegate selectors selected via runtime input.  
 
+### <+pipeline.storeType>
+
+If the pipeline is stored in Harness, the expression resolves to inline. If the pipeline is stored in a Git repository, the expression resolves to remote.
```
Comment (SudarshanPurohit) [inline docs/platform/variables-and-expressions/harness-variables.md:744]: maybe we add formatting around "inline" and "remote" so it's easier to understand?

Comment (SudarshanPurohit) [reply docs/platform/variables-and-expressions/harness-variables.md:744]: Also we should be adding formatting around the "null" in the second and third entries. 

Other than these, this looks good. Thanks, @sarthakkasat

Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-variables.md:744]:
```suggestion
If the pipeline is stored in Harness, the expression resolves to `inline`. If the pipeline is stored in a Git repository, the expression resolves to `remote`.
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 5]
```diff
@@ -739,6 +739,18 @@ The list of stages selected for execution.
...
+
+### <+pipeline.repoName>
+
+For remote pipelines, the expression resolves to the Git repository name. For inline pipelines, the expression resolves to null.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-variables.md:748]:
```suggestion
For remote pipelines, the expression resolves to the Git repository name. For inline pipelines, the expression resolves to `null`.
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 6]
```diff
@@ -739,6 +739,18 @@ The list of stages selected for execution.
...
+
+### <+pipeline.branchName>
+
+For remote pipelines, the expression resolves to the Git branch where the pipeline exists. For inline pipelines, the expression will resolve to null.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-variables.md:752]:
```suggestion
For remote pipelines, the expression resolves to the Git branch where the pipeline exists. For inline pipelines, the expression resolves to `null`.
```

**Review by brian-f-harness — APPROVED** [review]

**Review by brian-f-harness — APPROVED** [review]

**Pull request merged**